extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -47,61 +55,52 @@
+    public void configureDataSources(DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig,
+            DataSourcesJdbcBuildTimeConfig dataSourcesJdbcBuildTimeConfig,
+            DataSourcesRuntimeConfig dataSourcesRuntimeConfig,
+            DataSourcesJdbcRuntimeConfig dataSourcesJdbcRuntimeConfig,
+            boolean disableSslSupport) {
+        this.dataSourcesBuildTimeConfig = dataSourcesBuildTimeConfig;
+        this.dataSourcesJdbcBuildTimeConfig = dataSourcesJdbcBuildTimeConfig;
+        this.dataSourcesRuntimeConfig = dataSourcesRuntimeConfig;
+        this.dataSourcesJdbcRuntimeConfig = dataSourcesJdbcRuntimeConfig;
+        this.disableSslSupport = disableSslSupport;
     }
 
     public AgroalDataSource createDataSource(String dataSourceName,
             DataSourceBuildTimeConfig dataSourceBuildTimeConfig,
-            Optional<DataSourceRuntimeConfig> dataSourceRuntimeConfigOptional,
+            DataSourceJdbcBuildTimeConfig dataSourceJdbcBuildTimeConfig,
+            DataSourceRuntimeConfig dataSourceRuntimeConfig,
+            DataSourceJdbcRuntimeConfig dataSourceJdbcRuntimeConfig,
+            String resolvedDriveClass,
             boolean mpMetricsPresent) {
-        if (!dataSourceRuntimeConfigOptional.isPresent() || !dataSourceRuntimeConfigOptional.get().url.isPresent()) {
-            log.warn("Datasource " + dataSourceName + " not started: driver and/or url are not defined.");
+        checkConfigInjection();
+
+        if (!dataSourceJdbcBuildTimeConfig.enabled) {
+            return null;
+        }
+
+        if (!dataSourceJdbcRuntimeConfig.url.isPresent()) {
+            log.warn("JDBC datasource " + dataSourceName + " not started: driver and/or url are not defined.");
             return null;
         }
+
         // we first make sure that all available JDBC drivers are loaded in the current TCCL
         loadDriversInTCCL();
 
-        DataSourceRuntimeConfig dataSourceRuntimeConfig = dataSourceRuntimeConfigOptional.get();
-
-        String driverName = dataSourceBuildTimeConfig.driver.get();
         Class<?> driver;
         try {
-            driver = Class.forName(driverName, true, Thread.currentThread().getContextClassLoader());
+            driver = Class.forName(resolvedDriveClass, true, Thread.currentThread().getContextClassLoader());
         } catch (ClassNotFoundException e) {
-            throw new RuntimeException("Unable to load the dataSource driver", e);
+            throw new RuntimeException(
+                    "Unable to load the datasource driver " + resolvedDriveClass + " for datasource " + dataSourceName, e);
         }
 
-        String url = dataSourceRuntimeConfig.url.get();
+        InstanceHandle<AgroalConnectionConfigurer> agroalConnectionConfigurerHandle = Arc.container().instance(
+                AgroalConnectionConfigurer.class,
+                new JdbcDriverLiteral(dataSourceBuildTimeConfig.kind.get()));
```

Comment:
It seems like this hasn't been preceded by a `isPresent` anywhere.

Reply:
No, if I'm there, I'm sure it has been defined.

Reply:
OK